zkevm-circuits/src/evm_circuit/execution/storage/sload.rs

```diff
@@ -0,0 +1,378 @@
+
+    use bus_mapping::evm::OpcodeId;
+    use eth_types::{address, bytecode, evm_types::GasCost, ToLittleEndian, ToWord, Word};
+    use std::convert::TryInto;
+
+    fn test_ok(tx: eth_types::Transaction, key: Word, value: Word, is_warm: bool, result: bool) {
+        let rw_counter_end_of_reversion = if result { 0 } else { 19 };
+
+        let call_data_gas_cost = tx
+            .input
+            .0
+            .iter()
+            .fold(0, |acc, byte| acc + if *byte == 0 { 4 } else { 16 });
+
+        let randomness = rand_fp();
+        let bytecode = Bytecode::from(&bytecode! {
+            PUSH32(key)
+            #[start]
+            SLOAD
+            STOP
+        });
+        let block = Block {
+            randomness,
+            txs: vec![Transaction {
+                id: 1,
+                nonce: tx.nonce.try_into().unwrap(),
+                gas: tx.gas.try_into().unwrap(),
+                gas_price: tx.gas_price.unwrap_or_else(Word::zero),
+                caller_address: tx.from,
+                callee_address: tx.to.unwrap(),
+                is_create: tx.to.is_none(),
+                value: tx.value,
+                call_data: tx.input.to_vec(),
+                call_data_length: tx.input.0.len(),
+                call_data_gas_cost,
+                calls: vec![Call {
+                    id: 1,
+                    is_root: true,
+                    is_create: false,
+                    opcode_source: RandomLinearCombination::random_linear_combine(
+                        bytecode.hash.to_le_bytes(),
+                        randomness,
+                    ),
+                    result: Word::from(result as usize),
+                    rw_counter_end_of_reversion,
+                    is_persistent: result,
+                    callee_address: tx.to.unwrap(),
+                    ..Default::default()
+                }],
+                steps: vec![
+                    ExecStep {
+                        rw_indices: (0..8 + if result { 0 } else { 2 }).collect(),
+                        execution_state: ExecutionState::SLOAD,
+                        rw_counter: 9,
+                        program_counter: 33,
+                        stack_pointer: STACK_CAPACITY,
+                        gas_left: if is_warm {
+                            GasCost::WARM_STORAGE_READ_COST.as_u64()
+                        } else {
+                            GasCost::COLD_SLOAD_COST.as_u64()
+                        },
+                        gas_cost: if is_warm {
+                            GasCost::WARM_STORAGE_READ_COST.as_u64()
+                        } else {
+                            GasCost::COLD_SLOAD_COST.as_u64()
+                        },
+                        opcode: Some(OpcodeId::SLOAD),
+                        ..Default::default()
+                    },
+                    ExecStep {
+                        execution_state: ExecutionState::STOP,
+                        rw_counter: 17,
```

Comment:
should check `result` for next step `rw_counter `  ? just like above step `                        rw_indices: (0..8 + if result { 0 } else { 2 }).collect(),`

Reply:
I think we don't need to check for revert.

Like we don't check for revert in `BEGIN_TX`.

in `StepStateTransition` we use `rw_counter: Delta(8.expr()),`  and here we only have 1 case, instead of also considering revert case